terraso_backend/apps/shared_data/permission_rules.py

```diff
@@ -31,12 +31,20 @@ def allowed_to_delete_data_entry(user, data_entry):
 
 @rules.predicate
 def allowed_to_view_data_entry(user, data_entry):
-    return user.memberships.filter(group__in=data_entry.groups.all()).exists()
+    return (
+        user.memberships.approved_only()
+        .filter(group__in=data_entry.groups.all(), membership_status="approved")
```

Comment:
Do we need `membership_status="approved"` here since we have `approved_only()`?

Reply:
we don't that is redundant, I will remove that one